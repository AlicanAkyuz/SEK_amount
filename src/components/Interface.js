import React from 'react';
import PropTypes from 'prop-types';

import { FullWidthContainer, PrimaryText, SecondaryText } from '../styles';

const Interface = ({ primaryMessage, secondaryMessage }) => (
  <FullWidthContainer>
    <PrimaryText>{primaryMessage}</PrimaryText>
    <SecondaryText>{secondaryMessage}</SecondaryText>
  </FullWidthContainer>
);

Interface.propTypes = {
  primaryMessage: PropTypes.string.isRequired,
  secondaryMessage: PropTypes.string.isRequired,
};

export default Interface;
