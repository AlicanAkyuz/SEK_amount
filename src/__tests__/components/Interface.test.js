import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Interface from '../../components/Interface';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Interface primaryMessage="Loading..." secondaryMessage="The app is loading" />,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('When received a primary and a secondary text prop, it correctly renders a h3 tag for the primary and a h5 tag for the secondary text', () => {
  const tree = renderer
    .create(<Interface primaryMessage="Loading..." secondaryMessage="The app is loading" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
