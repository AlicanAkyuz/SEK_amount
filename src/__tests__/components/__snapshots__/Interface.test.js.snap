// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When received a primary and a secondary text prop, it correctly renders a h3 tag for the primary and a h5 tag for the secondary text 1`] = `
<div
  className="css-1a821q7"
>
  <h3
    className="css-qwi5oq"
  >
    Loading...
  </h3>
  <h5
    className="css-tgshpt"
  >
    The app is loading
  </h5>
</div>
`;
